Add a note about scaling above 100,000 registered users, reference loadtest framework (#4559)
mattermost/docs: source/deployment/scaling.rst

The following matrix presents key features for a successful multi-region Mattermost implementation that scales to 100,000 users with support for high availability and geographically based traffic routing in on premises, AWS, and Azure deployments.

> To scale to 100,000 users and above, it is recommended to use the Mattermost open source load testing framework to simulate usage of your system at full scale: https://github.com/mattermost/mattermost-load-test.

Reviewer: ```suggestion
To scale to 100,000 users and above, we recommended using `the Mattermost open source load testing framework <https://github.com/mattermost/mattermost-load-test.>`__ to simulate usage of your system at full scale.
```

> To scale to 100,000 users and above, we recommended using `the Mattermost open source load testing framework <https://github.com/mattermost/mattermost-load-test.>`__ to simulate usage of your system at full scale.

Reviewer 2: @jasonblais @cwarnermm The link here is broken.

Reviewer: Thanks @amyblais - Fixing via https://github.com/mattermost/docs/pull/4560
